File: CS/NewPageStarted/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace NewPageStarted {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        int pageCount = -1;
        public XtraReport1() {
            InitializeComponent();
        }

        private void Detail_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            if (PrintingSystem.Document.PageCount != pageCount) {
                lblNewPage.Visible = true;                
                pageCount = PrintingSystem.Document.PageCount;
                lblNewPage.Text = "New page: " + (pageCount + 1);
            }
            else {
                lblNewPage.Visible = false;
            }
        }

    }
}

File: VB/NewPageStarted/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace NewPageStarted
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Private pageCount As Integer = -1
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Detail_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles Detail.BeforePrint
			If PrintingSystem.Document.PageCount <> pageCount Then
				lblNewPage.Visible = True
				pageCount = PrintingSystem.Document.PageCount
				lblNewPage.Text = "New page: " & (pageCount + 1)
			Else
				lblNewPage.Visible = False
			End If
		End Sub

	End Class
End Namespace
